### PR TITLE
193 refactoring plan for workflow automation to berkley schema

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides:

 "TODO" comments in all of the `workflow_automation` packages intended to guide refactoring to be compatible with Berkley-style schema, database objects, and runtime API endpoints.  There are no code changes in this PR.

There are class and interaction diagrams based on analysis of the workflow_automation code base here:

https://drive.google.com/file/d/1ADrKHIDjN5JaaZSBXqqBuDOxiCLz7STk/view?usp=drive_link

### CODE REVIEW

File: nmdc_automation/workflow_automation/activities.py [hunk 1]
```diff
@@ -3,6 +3,13 @@
 from .workflows import Workflow
 from semver.version import Version
 
+# TODO: Berkley refactoring:
+#   The load_activities method will need to be modified to handle DataGeneration objects
```
Comment (aclum) [inline nmdc_automation/workflow_automation/activities.py:7]: DataGeneration is an abstract class, we should to its subclasses NucleotideSequencing or MassSpectrometry

File: nmdc_automation/workflow_automation/watch_nmdc.py [hunk 2]
```diff
@@ -14,7 +14,17 @@
 
 logger = logging.getLogger(__name__)
 
-
+# TODO: Berkley refactoring:
+#   The watcher interacts with the NMDC runtime API to find / claim jobs and to post the resulting
```
Comment (aclum) [inline nmdc_automation/workflow_automation/watch_nmdc.py:18]: any code that uses an endpoint that references activity is will change, see https://github.com/microbiomedata/nmdc-runtime/issues/593 for the new endpoint names. There is a linked draft PR which is not merged yet.

File: nmdc_automation/workflow_automation/wfutils.py [hunk 3]
```diff
@@ -11,7 +11,17 @@
 import hashlib
 from linkml_runtime.dumpers import json_dumper
 
-
+# TODO: Berkley refactoring:
+#  The NmdcSchema class - responsible for creating workflow and data object records
+#  to be inserted into the NMDC database will need to be updated to generate Berkley-compatible
+#  Datageneration and WorkflowExecution records.
```
Comment (aclum) [inline nmdc_automation/workflow_automation/wfutils.py:17]: Same comment here, DataGeneration is abstract.

**Review by aclum — COMMENTED** [review]
See comments on DataGeneration, also this PR is missing needed changes to config files. 
If used the files in test_data/metagenome_annotation_activity_set.json need to be updated to napa style identifiers and to have `type` defined.
configs/re_iding_worklfows.yaml should be deleted or updated 
updates are needed for the following files in the configs folder workflows.yaml, workflows-mt.yaml, import.yaml (see https://github.com/microbiomedata/nmdc_automation/issues/151 for example instructions)
python files in https://github.com/microbiomedata/nmdc_automation/blob/main/tests need to be updated, see https://github.com/microbiomedata/nmdc_automation/blob/main/tests/test_activities.py for example which refers to collections which won't exist anymore

File: nmdc_automation/workflow_automation/wfutils.py (hunk 3 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/wfutils.py:17]: Updated TODO with this detail

**Comment by aclum** [comment]
Here or in a different PR we also need to address TODOs for https://github.com/microbiomedata/nmdc_automation/blob/main/nmdc_automation/import_automation/activity_mapper.py

**Review by aclum — APPROVED** [review]

**Pull request merged**